Tropos/Controllers/TRAppDelegate.m

```diff
@@ -38,6 +40,32 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     return YES;
 }
 
+- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken;
+{
+    NSString *channel = [[NSTimeZone localTimeZone] name];
+    [self.courier subscribeToChannel:channel withToken:deviceToken];
+}
+
+- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler;
+{
+    if (![userInfo[@"aps"][@"content-available"] isEqual: @1]) {
+        completionHandler(UIBackgroundFetchResultNoData);
+        return;
+    }
+    [self.applicationController.conditionsDescription subscribeNext:^(NSAttributedString *conditions) {
```

Comment:
Here's what I mean:

```diff
-    [self.applicationController.conditionsDescription subscribeNext:^(NSAttributedString *conditions) {
+    [[self.applicationController.conditionsDescription take:1] subscribeNext:^(NSAttributedString *conditions) {
```

So the signal returned by the call to `take:` will immediately complete after the first value is sent. However, on thinking about it further, there's still a chance that this signal will never complete: in the event that an error occurs, no value will be sent on the `conditionsDescription` signal. This is because it's ultimately a KVO observation of a property that only changes when the weather update is successful.

In order to ensure that the fetch as actually successful, we need to do something like the following:

```objc
    RACSignal *updatedConditions = [[self.applicationController performBackgroundFetch] then:^{
        return [self.applicationController.conditionsDescription take:1];
    }];

    [updatedConditions subscribeNext:^(NSAttributedString *conditions) {
        // schedule local notification
    }];
```

Does that make sense?